dc/s2n-quic-dc/src/stream/server/tokio/tcp/worker.rs

```diff
@@ -116,7 +120,13 @@ where
         let prev_stream = core::mem::replace(&mut self.stream, Some((stream, remote_address)));
         let prev_ctx = core::mem::replace(&mut self.subscriber_ctx, Some(subscriber_ctx));
 
-        if let Some(remote_address) = prev_stream.map(|(_socket, remote_address)| remote_address) {
+        if let Some(remote_address) = prev_stream.map(|(socket, remote_address)| {
+            if linger.is_none() {
```

Comment:
```suggestion
            if linger == Some(Duration::ZERO) {
```

Maybe? Or we should document what we actually mean by None otherwise I suspect.

Reply:
Thinking about it more, it should be 

``` rust
if linger.is_none() || linger != Some(Duration::ZERO) {
```

Basically, we want to hard reset the connection if we drop an old stream for a new one. And the easiest way to do that is setting linger=0. So we only need to set the linger value to `0` if it wasn't already set to that.